### PR TITLE
SISRP-17838 - 5.4 - My Academics (Students) - Semesters - Enrollment Verification and Transcripts

### PR DESCRIPTION
**Pull request by redconfetti**
https://jira.berkeley.edu/browse/SISRP-17838

### CODE REVIEW

**Comment by redconfetti** [comment]
@christianvuerings - I took a before (left) and after (right) screenshot for the first commit where I removed the `cc-academics-semesters` style and replaced it with `cc-widget-padding`.

image

It looks a bit different, but I expect that I'll need to discuss the approach here with you. I'm trying to move this towards proper class re-use (`cc-widget-padding`), instead of repeating the same style of padding in specific styles like `cc-academic-semesters`.

Do you think there is some further diligence needed here?

**Comment by christianvuerings** [comment]
@redconfetti that's indeed why we had that style. In this particular case, I would verify with the designers whether they're ok with the visual change.

File: src/assets/templates/academics_semesters.html [hunk 1]
```diff
@@ -2,10 +2,15 @@
   <div class="cc-widget-title">
     <h2 class="cc-left">Semesters</h2>
     <a data-ng-href="{{transcriptLink}}" class="cc-right cc-widget-title-link" data-ng-if="transcriptLink">
-      <strong>Order Transcripts</strong>
+      <strong>Request Transcripts</strong>
     </a>
   </div>
-  <div class="cc-academics-semesters">
+  <div class="cc-widget-section">
```
Comment (christianvuerings) [inline src/assets/templates/academics_semesters.html]: @redconfetti Should we put the `data-ng-if="enrollmentVerificationLink"` on here or is this intentional? How does it look like when `enrollmentVerificationLink` is `false`?

Comment (redconfetti) [inline src/assets/templates/academics_semesters.html]: @christianvuerings Good call. I didn't think of that.

**Comment by redconfetti** [comment]
@christianvuerings

Do you think it would be valid to make a style like `cc-widget-padding` with padding on right and left, so that vertically stacked elements could have equal top and bottom padding, and thus not have that issue of extra padding above the first widget, and below the last widget?

**Comment by redconfetti** [comment]
Note: This task is not ready to merge yet.

**Comment by christianvuerings** [comment]
@redconfetti would suggest using `cc-widget-padding` as we do right now and have the vertical spacing as we talked about this morning https://github.com/ets-berkeley-edu/calcentral/blob/b3434526463a084f82d13bad25a47553cffa4ecd/src/assets/stylesheets/_flexbox.scss#L16

**Comment by redconfetti** [comment]
@christianvuerings Oooh, I totally missed the first-child / last-child logic when you were showing me that this morning. Thank You!!! 🍻

**Comment by raydavis** [comment]
Travis greenlighted this PR yesterday. I just restarted the Travis job, though, to confirm that the repeated failures on the next PR in line was not due to code changes on our end.

**Comment by raydavis** [comment]
The Travis re-run on this PR failed in the same way as @johncrossman 's later PR, https://github.com/ets-berkeley-edu/calcentral/pull/5270 :

```
> gulp build --env production

[16:15:11] Using gulpfile ~/build/ets-berkeley-edu/calcentral/gulpfile.js
[16:15:11] Starting 'build'...
[16:15:11] Starting 'build-clean'...
[16:15:11] Finished 'build-clean' after 36 ms
[16:15:11] Starting 'css'...
[16:15:12] Starting 'browserify'...
[16:15:12] Starting 'images'...
[16:15:12] Starting 'fonts'...

events.js:141
      throw er; // Unhandled 'error' event
      ^
Error: src/assets/stylesheets/_header.scss
Error: Mixins may not be defined within control directives or other mixins.
        on line 351 of src/assets/stylesheets/_header.scss
>>   @mixin cc-badges-small-screen($cc-badge-difference) {
   --^

    at options.error (/home/travis/build/ets-berkeley-edu/calcentral/node_modules/gulp-sass/node_modules/node-sass/lib/index.js:271:32)
```

**Comment by johncrossman** [comment]
@redconfetti Your build is failing on:
```
Error: Mixins may not be defined within control directives or other mixins.
        on line 351 of src/assets/stylesheets/_header.scss
>>   @mixin cc-badges-small-screen($cc-badge-difference) {
```
I recommend you rebase once #5270 is merged in order to pick up `_header.scss` fix.

**Comment by raydavis** [comment]
@redconfetti , is the visual change still in place? If so, have you checked with a designer about it? Thanks!

**Comment by redconfetti** [comment]
@raydavis I talked with Chara late yesterday and she was not able to provide any input on the issue. She did say that she will discuss it further with the functional owners. I'll be following up with her on that in [SISRP-18292](https://jira.berkeley.edu/browse/SISRP-18292)

**Comment by redconfetti** [comment]
@raydavis Don't merge yet. My browser refreshed and something else is askew with the design now.

**Comment by raydavis** [comment]
At the JIRA ticket, Chara wrote that "the spacing is acceptable." Thanks for checking, @redconfetti .

**Pull request merged**